### ISSUE TITLE
Make it easier to copy effects/customs

### ISSUE THREAD

**Issue by WolfspiritM** [CoraleStudios/Colore#156]
Right now when creating a Custom you can only set it to a specific Color.
It can be useful to copy an existing Custom in some cases to have a copy of the current state.

For example with a copy constructor:
```
var keyboard_custom = KeyboardCustom.Create();
keyboard_custom.Set(Color.Red);

var other_custom = new KeyboardCustom(keyboard_custom);
// Or maybe
var other_custom_2 = KeyboardCustom.Create(keyboard_custom);
```

**Comment by Sharparam** [comment]
Adding copy constructors is a good idea that will also be familiar for those used to .NET development. `Clone` methods could also be good to add which also follows what a lot of existing .NET code does.

**Comment by Sharparam** [comment]
Implemented in #158
